libraries/bot-connector/src/main/java/com/microsoft/bot/connector/authentication/OAuthClient.java

```diff
@@ -0,0 +1,344 @@
+//        {
+//        tokenUrl = tokenUrl.Replace("{magicCodeParam}", $"&code={Uri.EscapeDataString(magicCode)}");
+//        }
+//        else
+//        {
+//        tokenUrl = tokenUrl.Replace("{magicCodeParam}", String.Empty);
+//        }
+
+        // Create HTTP transport objects
+//        var httpRequest = new HttpRequestMessage();
+//        HttpResponseMessage httpResponse = null;
+//        httpRequest.Method = new HttpMethod("GET");
+//        httpRequest.RequestUri = new Uri(tokenUrl);
+
+        // add botframework api service url to the list of trusted service url's for these app credentials.
+//        MicrosoftAppCredentials.trustServiceUrl(tokenUrl);
+
+        // Set Credentials
+//        if (_client.Credentials != null)
+//        {
+//        cancellationToken.ThrowIfCancellationRequested();
+//        await(_client.Credentials.ProcessHttpRequestAsync(httpRequest));
+//        }
+//        cancellationToken.ThrowIfCancellationRequested();
+
+//        if (shouldTrace)
+//        {
+//         ServiceClientTracing.SendRequest(invocationId, httpRequest);
+//        }
+//        httpResponse = await(_client.HttpClient.SendAsync(httpRequest, cancellationToken));
+//        if (shouldTrace)
+//        {
+//         ServiceClientTracing.ReceiveResponse(invocationId, httpResponse);
+//        }
+//        HttpStatusCode statusCode = httpResponse.StatusCode;
+//        cancellationToken.ThrowIfCancellationRequested();
+//        if (statusCode == HttpStatusCode.OK)
+//        {
+//         String responseContent = await(httpResponse.Content.ReadAsStringAsync());
+//        try
+//        {
+//        var tokenResponse = Rest.Serialization.SafeJsonConvert.DeserializeObject<TokenResponse>(responseContent);
+//        return tokenResponse;
+//        }
+//        catch (JsonException)
+//        {
+//        // ignore json exception and return null
+//        httpRequest.Dispose();
+//        if (httpResponse != null)
+//        {
+//        httpResponse.Dispose();
+//        }
+        return null;
+    }
+    //}
+//        else if (statusCode == HttpStatusCode.NotFound)
+//        {
+//        return null;
+//        }
+//        else
+//        {
+//        return null;
+//        }
+//        }
+
+    /// <summary>
+/// Signs Out the User for the given ConnectionName.
+/// </summary>
+/// <param name="userId"></param>
+/// <param name="connectionName"></param>
+/// <param name="cancellationToken"></param>
+/// <returns></returns>
+    public CompletableFuture<Boolean> SignOutUserAsync(String userId, String connectionName) {
+        if (StringUtils.isEmpty(userId)) {
+            throw new IllegalArgumentException("userId");
+        }
+        if (StringUtils.isEmpty(connectionName)) {
+            throw new IllegalArgumentException("connectionName");
+        }
+
+        //boolean shouldTrace = // ServiceClientTracing.IsEnabled;
+        String invocationId = null;
+//        if (shouldTrace)
```

Comment:
presumably a work in progress??? hence the review title "skeleton"

Reply:
Yes, much more to convert before I can fully light this up